### PR TITLE
#313 Fix invalid behavior for `parameterized="false"` in MFTF

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description
Fix for reported `parameterized="false"` behavior in Page Object.
Applied also to Section Object

### Fixed Issues (if relevant)
1. magento/magento2-functional-testing-framework#313

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: src/Magento/FunctionalTestingFramework/Page/Handlers/SectionObjectHandler.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ private function __construct()
                     $elementSelector = $elementData[SectionObjectHandler::SELECTOR] ?? null;
                     $elementLocatorFunc = $elementData[SectionObjectHandler::LOCATOR_FUNCTION] ?? null;
                     $elementTimeout = $elementData[SectionObjectHandler::TIMEOUT] ?? null;
-                    $elementParameterized = $elementData[SectionObjectHandler::PARAMETERIZED] ?? false;
+                    $elementParameterized = $this->getBoolean($elementData[SectionObjectHandler::PARAMETERIZED] ?? null);
```
Comment (lbajsarowicz) [inline src/Magento/FunctionalTestingFramework/Page/Handlers/SectionObjectHandler.php]: `?? null` avoids throwing a warning about the reference to the unknown key.

File: src/Magento/FunctionalTestingFramework/Page/Handlers/PageObjectHandler.php [hunk 2]
```diff
@@ -66,7 +66,7 @@ private function __construct()
 
             $module = $pageData[self::MODULE] ?? null;
             $sectionNames = array_keys($pageData[self::SECTION] ?? []);
-            $parameterized = $pageData[self::PARAMETERIZED] ?? false;
+            $parameterized = $this->getBoolean($pageData[self::PARAMETERIZED] ?? null);
```
Comment (lbajsarowicz) [inline src/Magento/FunctionalTestingFramework/Page/Handlers/PageObjectHandler.php]: `?? null` avoids throwing a warning about the reference to the unknown key.

**Comment by lbajsarowicz** [comment]
> Consider utilizing the default value passed to the getBoolean method.

Actually this attribute is redundant. Eventually, we should introduce DataObject to carry such DTO.

**Comment by KevinBKozan** [comment]
@lbajsarowicz Thank you for your contribution! I've created an internal PR to track progress on revieweing this, I'll keep you updated with comments here.

**Comment by lbajsarowicz** [comment]
MFTF team changed the approach to `parameterized` attribute.

**Pull request closed without merging**